homeassistant/components/zwave_js/sensor.py

```diff
@@ -241,16 +326,26 @@ def __init__(
         self,
         config_entry: ConfigEntry,
         client: ZwaveClient,
-        info: ZwaveDiscoveryInfo,
+        entity_description: ZwaveSensorEntityDescription,
     ) -> None:
         """Initialize a ZWaveNumericSensor entity."""
-        super().__init__(config_entry, client, info)
+        super().__init__(config_entry, client, entity_description)
 
         # Entity class attributes
         self._attr_state_class = STATE_CLASS_MEASUREMENT
         if self.device_class == DEVICE_CLASS_ENERGY:
             self._attr_last_reset = dt.utc_from_timestamp(0)
```

Comment:
Can we move this to the entity description?

Reply:
Can you explain what we were trying to achieve here? I made this change, but for other meter types the last reset will still get set for other meter types when the reset_meter service is called

Reply:
We have a guard in place here:
https://github.com/home-assistant/core/blob/7590cb2861f55d72c5f7b459e78d4f9f01b7a50c/homeassistant/components/zwave_js/sensor.py#L277-L279

Reply:
Only entities that represent accumulating measurements should set `last_reset` attribute.

Reply:
I tried creating a base class for them to be subclasses of too to use that as the type but that also threw up errors

Reply:
Also tried IntEnum 🙂

Reply:
I'm not following regarding what part of the code was attempted to be changed.

Reply:
Sorry, that was for another PR 🤦🏾

Reply:
ah, now I get it, we never connect to the signal if we return there, thus skipping the update. 👍🏾